be/src/runtime/types.cpp

```diff
@@ -264,9 +264,9 @@ TypeDescriptor::TypeDescriptor(const google::protobuf::RepeatedPtrField<PTypeNod
     }
 }
 
-void TypeDescriptor::add_sub_type(TypeDescriptor&& sub_type, bool&& is_nullable) {
-    children.emplace_back(sub_type);
-    contains_nulls.emplace_back(is_nullable);
+void TypeDescriptor::add_sub_type(TypeDescriptor sub_type, bool is_nullable) {
```

Comment:
`OrcReader` has used this method, and use right values as parameters.

Reply:
This function can match right values, and if the argument passed in is a right value, the overhead is two move operations

Reply:
OK, it looks well. How about another polymorphic function `add_sub_type(TypeDescriptor&& sub_type, std::string&& field_name, bool&& is_nullable)`? It feels strange that the parameter forms of the two functions are not uniform.